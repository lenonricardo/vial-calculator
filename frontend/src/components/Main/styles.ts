import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: var(--theme-background);
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
`

export const Calculator = styled.div`
  width: 450px;
  height: 750px;
  background-color: #fff;
  border-radius: 1.5rem;
  box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
`

export const PadButtons = styled.div`
  width: 450px;
  height: 450px;
  border-radius: 0 0 1.5rem 1.5rem;
  display: f;
  flex-direction: row;
`

export const Display = styled.div`
  width: 450px;
  height: 300px;
  border-radius: 1.5rem 1.5rem 0 0;
`

export const BottomButtons = styled.div`
  display: flex;
  flex-direction: row;
`

export const NumberButtons = styled.div`
  display: flex;
  flex-direction: column;
`

export const SecondaryButtons = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(90px, 1fr));
  grid-template-rows: repeat(auto-fill, minmax(90px, 1fr));
  width: calc(450px - 90px - 90px);
  max-height: calc(450px - 90px);
`

export const OperationButtons = styled.div`
  display: flex;
  flex-direction: column;
`

export const TerciaryButtons = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
`

export const Button = styled.div`
  width: 90px;
  height: 90px;
  color: var(--font-pads);
  display: flex;
  font-size: 35px;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  user-select: none;

  transition: background 0.8s;

  /* &:hover {
    background: var(--secondary-operations) radial-gradient(circle, transparent 1%, var(--secondary-operations) 1%) center/15000%;
  }

  &:active {
    background-color: var(--secondary-operations);
    background-size: 100%;
    transition: background 0s;
  } */
`

export const ButtonSecondary = styled(Button)`
  background-color: var(--secondary-operations);
`

export const DoubleButton = styled(Button)`
  width: calc(90px + 90px);
`

export const ButtonPrimary = styled(Button)`
  background: var(--theme-primary);
  color: #fff;
`
export const ButtonGradient1 = styled(Button)`
  background: linear-gradient(
    180deg,
    var(--theme-primary) 9%,
    var(--theme-secondary) 100%
  );
  color: #fff;
`

export const ButtonGradient2 = styled(Button)`
  background: var(--theme-secondary);
  color: #fff;

  &:last-child {
    border-radius: 0 0 1.5rem 0;
  }
`
