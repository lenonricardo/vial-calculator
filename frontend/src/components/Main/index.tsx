import * as S from './styles'

const Main = ({}) => (
  <S.Wrapper>
    <S.Calculator>
      <S.Display></S.Display>
      <S.PadButtons>
        <S.NumberButtons>
          <S.SecondaryButtons>
            <S.ButtonSecondary>CE</S.ButtonSecondary>
            <S.ButtonSecondary>C</S.ButtonSecondary>
            <S.ButtonSecondary>+/-</S.ButtonSecondary>

            <S.Button>7</S.Button>
            <S.Button>8</S.Button>
            <S.Button>9</S.Button>
            <S.Button>4</S.Button>
            <S.Button>5</S.Button>
            <S.Button>6</S.Button>
            <S.Button>1</S.Button>
            <S.Button>2</S.Button>
            <S.Button>3</S.Button>
          </S.SecondaryButtons>

          <S.BottomButtons>
            <S.DoubleButton>0</S.DoubleButton>
            <S.Button>.</S.Button>
          </S.BottomButtons>
        </S.NumberButtons>

        <S.TerciaryButtons>
          <S.ButtonSecondary>%</S.ButtonSecondary>
          <S.ButtonSecondary>(</S.ButtonSecondary>
          <S.ButtonSecondary>)</S.ButtonSecondary>
          <S.ButtonSecondary>√</S.ButtonSecondary>
          <S.ButtonSecondary>x²</S.ButtonSecondary>
        </S.TerciaryButtons>

        <S.OperationButtons>
          <S.ButtonPrimary>÷</S.ButtonPrimary>
          <S.ButtonGradient1>x</S.ButtonGradient1>
          <S.ButtonGradient2>-</S.ButtonGradient2>
          <S.ButtonGradient2>+</S.ButtonGradient2>
          <S.ButtonGradient2>=</S.ButtonGradient2>
        </S.OperationButtons>
      </S.PadButtons>
    </S.Calculator>
  </S.Wrapper>
)

export default Main
