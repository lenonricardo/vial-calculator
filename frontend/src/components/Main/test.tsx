import { render } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the component', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
