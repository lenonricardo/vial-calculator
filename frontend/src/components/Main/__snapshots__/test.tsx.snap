// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: var(--theme-background);
  width: 100%;
  height: 100%;
}

<main
  class="c0"
/>
`;
